src/SerializeTool.cpp

```diff
@@ -0,0 +1,479 @@
+
+        if (result)
+            std::cout << result->getJson(0).toStyledString() << std::endl;
+        else
+            fail(boost::none);
+    }
+    catch (std::exception const& e)
+    {
+        fail(e);
+    }
+}
+
+void
+doSingleSign(std::string const& data,
+    boost::filesystem::path const& keyFile)
+{
+    using namespace serialize;
+    auto const fail = [&](boost::optional<std::exception const&> e)
+    {
+        std::cout << "Unable to sign \"" << data << "\"" << std::endl;
+        if (e)
+            std::cout << "Reason: " << e->what() << std::endl;
+    };
+    try
+    {
+        auto tx = make_sttx(boost::trim_copy(data));
+        if (tx)
+        {
+            auto const rippleKey = RippleKey::make_RippleKey(keyFile);
+
+            rippleKey.singleSign(*tx);
+
+            std::cout << tx->getJson(0).toStyledString() << std::endl;
+        }
+        else
+            fail(boost::none);
+    }
+    catch (std::exception const& e)
+    {
+        fail(e);
+    }
+}
+
+void
+doMultiSign(std::string const& data,
+    boost::filesystem::path const& keyFile)
+{
+    using namespace serialize;
+    auto const fail = [&](boost::optional<std::exception const&> e)
+    {
+        std::cout << "Unable to sign \"" << data << "\"" << std::endl;
+        if (e)
+            std::cout << "Reason: " << e->what() << std::endl;
+    };
+    try
+    {
+        auto tx = make_sttx(boost::trim_copy(data));
+        if (tx)
+        {
+            auto rippleKey = RippleKey::make_RippleKey(keyFile);
+
+            rippleKey.multiSign(*tx);
+
+            std::cout << tx->getJson(0).toStyledString() << std::endl;
+        }
+        else
+            fail(boost::none);
+    }
+    catch (std::exception const& e)
+    {
+        fail(e);
+    }
+}
+
+void
+doCreateKeyfile(boost::filesystem::path const& keyFile,
+    boost::optional<std::string> const& keytype,
+    boost::optional<std::string> const& seed)
+{
+    using namespace ripple;
+    using namespace serialize;
+
+    if (exists(keyFile))
+        throw std::runtime_error(
+            "Refusing to overwrite existing key file: " +
+            keyFile.string());
+
+    auto const kt = keytype ?
+        boost::optional<KeyType>{keyTypeFromString(*keytype)} :
+        boost::none;
+    if (kt && *kt == KeyType::invalid)
+    {
+        std::cout << "Invalid key type: \"" << *keytype << "\"" <<
+            std::endl;
+        return;
+    }
+
+    auto key = RippleKey::make_RippleKey(kt, seed);
+
+    key.writeToFile(keyFile);
+
+    std::cout << "New ripple key created " <<
+        "in " << keyFile.string() << ".\n" <<
+        "Key type is " << to_string(key.keyType()) << ", and " <<
+        "account ID is " <<
+        toBase58(calcAccountID(key.publicKey())) << ".\n" <<
+        "\nThis file should be stored securely and not shared.\n\n";
+
+}
+
+std::string
+getStdin()
+{
+    std::ostringstream stdinput;
+    stdinput << std::cin.rdbuf();
```

Comment:
Depends on your OS. Under unix-like, hit `enter`, then `ctrl-d`. Under a true Windows shell (ie. `cmd.exe`), hit `enter`, then `ctrl-z`, then `enter` again.

Unfortunately, running the Windows build in a unix-like shell, requires mixing the metaphors, because the Windows app doesn't recognize `ctrl-d`, but `ctrl-z` will stop the job. In that case, run
```
$ cat | ripple-serialize deserialize
```
And treat it like unix. This isn't a bug in the app, it's just the way the shells interact.